### PR TITLE
GPS failsafes for SiteScan

### PR DESCRIPTION
**Pull request by blandry**
This is a refactor of our handling of the GPS/Position estimate in SiteScan. Some of the notable changes are that the copter will not switch to ALT_HOLD but will RTL instead if it regains GPS while landing (in a mission). There is a lot to this PR that is safety critical so perhaps two reviewers would be better (@wsilva32 could you also review?). Please refer to the google doc on this topic. I will not link to it on this PR because this is a public repo but there is a link to it in the associated Jira ticket. The document also contains a list of the tests I have performed so far.

### CODE REVIEW

File: ArduCopter/ekf_check.pde [hunk 1]
```diff
@@ -133,17 +135,26 @@ static void failsafe_ekf_event()
     failsafe.ekf = true;
     Log_Write_Error(ERROR_SUBSYSTEM_FAILSAFE_EKFINAV, ERROR_CODE_FAILSAFE_OCCURRED);
 
-    // take action based on fs_ekf_action parameter
-    switch (g.fs_ekf_action) {
-        case FS_EKF_ACTION_ALTHOLD:
-            // AltHold
-            if (failsafe.radio || !set_mode(ALT_HOLD)) {
+    // make sure the gps glitch failsafe does nothing on resilve
```
Comment (wsilva32) [inline ArduCopter/ekf_check.pde]: comment typo

File: libraries/SITL/SITL.h [hunk 2]
```diff
@@ -98,6 +98,8 @@ class SITL
     AP_Int16  mag_delay; // magnetometer data delay in ms
     AP_Int16  wind_delay; // windspeed data delay in ms
 
+    AP_Int16 gps_spd_err; // reported GPS speed accuracy in cm/s
```
Comment (wsilva32) [inline libraries/SITL/SITL.h]: a bit of a confusing variable name. maybe it should gps_spd_acc so people don't think it's a controller error value?

File: libraries/AP_HAL_AVR_SITL/sitl_gps.cpp [hunk 3]
```diff
@@ -245,7 +245,7 @@ void SITL_State::_update_gps_ubx(const struct gps_data *d)
 	if (velned.heading_2d < 0.0f) {
 		velned.heading_2d += 360.0f * 100000.0f;
 	}
-	velned.speed_accuracy = 40;
+    velned.speed_accuracy = _sitl->gps_spd_err;
```
Comment (wsilva32) [inline libraries/AP_HAL_AVR_SITL/sitl_gps.cpp]: accidental indent?

File: libraries/SITL/SITL.h (hunk 2 above)
Comment (blandry) [inline libraries/SITL/SITL.h]: I'll let @priseborough comment on this. I think the controller variables have quite different names though.

**Comment by blandry** [comment]
@priseborough did you get a chance to look at this?

File: libraries/SITL/SITL.h (hunk 2 above)
Comment (priseborough) [inline libraries/SITL/SITL.h]: It doesn't clash with any controller name

**Comment by priseborough** [comment]
I had trouble reviewing it as separate commits when I first looked at it last week, so will set aside time today to review the complete files.

**Comment by priseborough** [comment]
I've done a code walkthrough and am satisfied that the changes are consistent with the requirements document. I recommend this is pulled in.

**Pull request merged**